CONTRIBUTING.md

```diff
@@ -149,335 +231,329 @@ and dependencies.
+```
 
-You can run your workflow using the `./run launch` command:
+**For Workflows:**
 
-```bash
-./run launch workflows/my_workflow
+```cmake
+# In ./workflow/CMakeLists.txt
+add_holohub_application(my_workflow DEPENDS
+                        OPERATORS my_operator1 my_operator2)
 ```
 
-### Adding a Package Configuration
-
-1. Ensure the applications/operator CMake targets and/or files you want to package have an `install` rule define in your application/operator CMakeLists.txt files (see [CMake docs](https://cmake.org/cmake/help/latest/command/install.html)). Optionally, pass a `COMPONENT` argument to the `install` rule to control precisely which of your targets/files will get packaged. This can be useful if - for example - you want to create dev vs run packages, not include python bindings, or split your libraries in separate packages for different backends.
-2. Create a folder for your package (or group of packages) under the [`pkg`](./pkg/) directory.
-3. Create a `CMakeLists.txt` file in that new folder with the following contents:
-
-   ```cmake
-   holohub_configure_deb(
-     NAME "my-package-dev"
-     COMPONENTS "my-headers" "my-libs" # (optional) list of installation components to package, Default: all targets/files configured with an install rule will be packaged.
-     DESCRIPTION "My project (dev)"
-     VERSION "X.Y.Z"
-     VENDOR "My org name"
-     CONTACT "John Doe <john@doe.org>"
-     DEPENDS "libc6 (>= 2.34), libstdc++6 (>= 11)" # list of your package debian dependencies - https://www.debian.org/doc/debian-policy/ch-relationships.html
-     SECTION "devel" # (optional)
-     PRIORITY "optional" # (optional)
-   )
-   ```
+**For Packages:**
 
-4. Update the `CMakeLists.txt` file under [`pkg`](./pkg/CMakeLists.txt) to include this new package directory and define the app and operators to build and package using the `add_holohub_package` CMake function:
+**CMake Configuration:**
 
-   ```bash
-   add_holohub_package(my_packager
-                       APPLICATIONS my_app1
-                       OPERATORS my_op1 my_op2)
-   ```
+```cmake
+# In ./pkg/my_package/CMakeLists.txt
+holohub_configure_deb(
+  NAME "my-package-dev"
+  COMPONENTS "my-headers" "my-libs"  # optional
+  DESCRIPTION "My project description"
+  VERSION "1.0.0"
+  VENDOR "Your Organization"
+  CONTACT "Your Name <your.email@example.com>"
+  DEPENDS "libc6 (>= 2.34), libstdc++6 (>= 11)"
+  SECTION "devel"      # optional
+  PRIORITY "optional"  # optional
+)
+```
 
-5. You can then generate the package(s) by configuring your build with `-D PKG_my_packager:BOOL=1` :
+**Package Registration:**
 
-    ```bash
-    cmake -S . -B ./build -D PKG_my_packager:BOOL=1 # NOTE: avoid adding any other -D PKG_, -D APP_ or -D OP_ as their built targets would currently pollute the content of your package, unless you used COMPONENTS in holohub_configure_deb above
-    cmake --build ./build -j
-    cpack --config ./build/pkg/CPackConfig-*.cmake
-    ```
```

Comment:
Could we migrate advance usage documentation with CMake to developer docs: https://github.com/nvidia-holoscan/holohub/blob/main/doc/developer.md ?

Reply:
Thanks for your suggestion, @tbirdso! It seems that most CMake commands in this documentation are already covered in the `developer.md` (excluding cpack). However, that document also requires an update. I’ll create a follow-up PR to address that.